hail/src/main/scala/is/hail/expr/types/physical/PStruct.scala

```diff
@@ -28,7 +28,7 @@ object PStruct {
   def canonical(t: PType): PStruct = PCanonicalStruct.canonical(t)
 }
 
-abstract class PStruct extends PBaseStruct {
+trait PStruct extends PBaseStruct {
```

Comment:
may be better to keep as abstract class if has implementations, in line with pure interfaces in Java.

Reply:
ah for mixin, hm, ok